.github/workflows/release.yml

```diff
@@ -0,0 +1,34 @@
+name: Release
+
+# This workflow follows Pypi guidelines for publishing using trusted publishers
+# See https://docs.pypi.org/trusted-publishers/using-a-publisher/ for more details.
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  package:
+    name: Package
+    uses: ./.github/workflows/package.yml
+
+  release:
+    name: Release
+    needs: [package]
+    runs-on: ubuntu-latest
+
+    # Make the GH environment explicit
+    environment: release
+
+    # Mandatory for trusted publishing
+    permissions:
+      id-token: write
```

Comment:
Today I migrated the windows-curses repo to use the "trusted publisher" mechanism and found that [`contents: write` is also required when triggered on "release published" event](https://github.com/zephyrproject-rtos/windows-curses/blob/61281da18178d49d505be2ec285feee9a776ccad/.github/workflows/release.yml#L19).

Reply:
Updated

Reply:
I think this was because of the release assets being uploaded using the `softprops/action-gh-release` action.

But you make a good point, I've added this step.